pkg/client/typed/generated/autoscaling/unversioned/scale_expansion.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/autoscaling"
+)
+
+// The ScaleExpansion interface allows manually adding extra methods to the ScaleInterface.
+type ScaleExpansion interface {
```

Comment:
The clientset currently includes client for scale just to mimic what the unversioned client provides: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/scale.go

I agree with @lavalamp that we should treat it at a subresource, like how we treat `status`. I'm just not sure if the apiserver expects the `scale` to be encoded with apiVersion=`extensions/v1beta1`, if not, this should be easy to fix.